tools/mmp/aot.cs

```diff
@@ -0,0 +1,193 @@
+					continue;
+				}
+				if (option.StartsWith ("-", StringComparison.Ordinal)) {
+					if (aotType == AotType.Default)
+						aotType = AotType.Explicit;
+					excludedAssemblies.Add (option.Substring (1), false);
+					continue;
+				}
+				throw new MonoMacException (20, true, "The valid options for '{0}' are '{1}'.", "--aot", "none, all, core, sdk, and an explicit list of assemblies.");
+			}
+			if (aotType == AotType.Default)
+				throw new MonoMacException (20, true, "The valid options for '{0}' are '{1}'.", "--aot", "none, all, core, sdk, and an explicit list of assemblies.");
+		}
+
+		List<string> GetFilesToAOT (IFileEnumerator files)
+		{
+			var aotFiles = new List<string> ();
+			foreach (var file in files.Files) {
+				string fileName = Path.GetFileName (file);
+				string extension = Path.GetExtension (file);
+				if (extension != ".exe" && extension != ".dll")
+					continue;
+
+				if (excludedAssemblies.ContainsKey (fileName)) {
+					excludedAssemblies[fileName] = true;
+					continue;
+				}
+
+				if (includedAssemblies.ContainsKey (fileName)) {
+					includedAssemblies[fileName] = true;
+					aotFiles.Add (file);
+					continue;
+				}
+
+				switch (aotType) {
+				case AotType.All:
+					aotFiles.Add (file);
+					break;
+				case AotType.SDK:
+					string fileNameNoExtension = Path.GetFileNameWithoutExtension (fileName);
+					if (Profile.IsSdkAssembly (fileNameNoExtension) || fileName == "Xamarin.Mac.dll")
+						aotFiles.Add (file);
+					break;
+				case AotType.Core:
+					if (fileName == "Xamarin.Mac.dll" || fileName == "System.dll" || fileName == "mscorlib.dll")
+						aotFiles.Add (file);
+					break;
+				case AotType.Explicit:
+					break; // In explicit, only included includedAssemblies included
+				default:
+					throw ErrorHelper.CreateError (0099, "Internal error \"GetFilesToAOT with aot: {0}\" Please file a bug report with a test case (http://bugzilla.xamarin.com).", aotType);
+				}
+			}
+
+			var unusedIncludes = includedAssemblies.Where (pair => !pair.Value).Select (pair => pair.Key).ToList ();
+			if (unusedIncludes.Count > 0)
+				throw ErrorHelper.CreateError (3009, "AOT of '{0}' was requested but was not found", String.Join (" ", unusedIncludes));
+
+			var unusedExcludes = excludedAssemblies.Where (pair => !pair.Value).Select (pair => pair.Key).ToList ();
+			if (unusedExcludes.Count > 0)
+				throw ErrorHelper.CreateError (3010, "Exclusion of AOT of '{0}' was requested but was not found", String.Join (" ", unusedExcludes));
+
+			return aotFiles;
+		}
+
+		public void Compile (string path)
+		{
+			Compile (new FileSystemEnumerator (path));
+		}
+
+		public void Compile (IFileEnumerator files)
+		{
+			if (!IsAOT)
+				throw ErrorHelper.CreateError (0099, "Internal error \"AOTBundle with aot: {0}\" Please file a bug report with a test case (http://bugzilla.xamarin.com).", aotType);
+
+			string monoExePath = Path.Combine (XamarinMacPrefix, "bin/bmac-mobile-mono" + (Is64Bit ? "" : "-32"));
```

Comment:
Does this work for all the three profiles (mobile, xm 4.5 and system)? I believe we'll have to use the system mono when using the system profile.